### DISCUSSION TITLE
Issue setting up hardware acceleration with Nvidia

### DISCUSSION THREAD

**Discussion by micogg** [blakeblackshear/frigate#2547] [Ask A Question]
I'm running Frigate in a TrueNAS Scale docker with the image " blakeblackshear/frigate:stable-amd64nvidia ".
I decided to try using Nvidia hardware acceleration as I had a 1060 sitting lifeless in the case. 
TrueNAS is set to pass the GPU resource to the container including the variables " NVIDIA_VISIBLE_DEVICES=all " and " NVIDIA_DRIVER_CAPABILITIES=compute,utility,video ".  

I am able to see the GPU from within the container, but it is showing no processes running.  

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.91.03    Driver Version: 460.91.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce GTX 106...  Off  | 00000000:10:00.0 Off |                  N/A |
| 38%   39C    P2    30W / 120W |    552MiB /  3016MiB |      7%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
+-----------------------------------------------------------------------------+

I am also not sure that the codecs showing are the Nvidia codecs.

 V..... h264_cuvid            (codec h264)
 V..... hevc_cuvid            (codec hevc)
 V..... mjpeg_cuvid           (codec mjpeg)
 V..... mpeg1_cuvid           (codec mpeg1video)
 V..... mpeg2_cuvid           (codec mpeg2video)
 V..... mpeg4_cuvid           (codec mpeg4)
 V..... vc1_cuvid             (codec vc1)
 V..... vp8_cuvid             (codec vp8)
 V..... vp9_cuvid             (codec vp9)

I have the Frigate config set to the below arguments.

cameras:
  back:
    ffmpeg:
      hwaccel_args:
        - -c:v
        - h264_cuvid
      inputs:
        - path: rtsp://admin:TLVSucks@192.168.1.154:554//h264Preview_01_main
          roles:
            - detect
            - rtmp
            - record
    detect:
      width: 1280
      height: 720

Any advice on what I'm missing here to enable the Nvidia hardware acceleration would be greatly appreciated.

**Comment by stephack** [comment]
@micogg 
Seems to me like you already have it working. I see the same result from the nvidia-smi shell command in my plex and frigate containers. For some reason the last line in the output seems to get truncated when viewed from within the container. If you run the nvidia-smi command from the TrueNas shell (System Settings->Shell) , it shows that ffmeg process is using the card. 
I see the exact same behavior with my Plex container when it's transcoding (ie last line of output is only visible from the TrueNas Shell).

**Comment by bugui2007** [comment]
I have the same issue. did you check ls -la /dev/dri to know if you have a render group on it? (or just 109)
